adobe/spectrum-web-components
# fix(number-field): allow only numeric characters for Japanese/Chinese IME

## Description

Text candidates appear for characters by inputting them with Japanese/Simplified Chinese/Traditional Chinese IME for numbers in respective fields where only digits are allowed.

## Related issue(s)



- #4805 

## Motivation and context

I think users should not be allowed to type "love" for example in an `sp-number-field`, when using Japanese/Chinese IME.
We do this check for english keyboard, but it seems it was missed to internationalised keyboards. 
Internal tracking: CCEX-145560

## How has this been tested?




-   [ ] _Test case 1_ Repeat for Safari/Chrome/Firefox
    1. Go [here](https://rocss-4805--spectrum-web-components.netlify.app/components/number-field/)
    2. Make sure to use Japanese IME (you can find this in the Keyboard settings)
    3. Try to enter some text (ex. 'あい'in JP) into the numeric input box.
    4. You should not be allowed to enter characters that are not numeric


## Screenshots (if appropriate)

## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
